app/src/main/java/com/toolslab/noisepercolator/db/DataProvider.kt

```diff
@@ -19,36 +22,88 @@ class DataProvider(private val context: Context = NoisePercolator.applicationCon
 
         @VisibleForTesting
         const val MESSAGES_KEY = "MESSAGES_KEY"
+
+        const val SHARED_PREFERENCES_DEPRECATED_MESSAGE = "Now using Realm instead of SharedPreferences."
     }
 
     @CheckResult
-    fun getMessages(): List<Message> {
+    fun getMessages(): Observable<List<Message>> {
+        migrateOldDatabaseIfNeeded()
+        val realm = realmWrapper.getDefaultInstance()
+        return realm.where(Message::class.java).findAllAsync()
```

Comment:
Mhhh yes I'm wondering if this would be better or if it is equally hard to read ;)

```kotlin
        return realmWrapper.getDefaultInstance().run {
            this.where(Message::class.java)
                    .findAllAsync()
                    .asFlowable()
                    .toObservable()
                    .map {
                        this.copyFromRealm(it)
                    }
                    .doOnDispose {
                        this.close()
                    }
                    .doOnError {
                        Timber.e(it, "Error in DataProvider.getMessages()")
                    }
        }
```

What would you choose, run or apply?

Reply:
`run`, as you do not need the return value